life_expectancy/tests/test_cleaning.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for the cleaning module"""
+import pandas as pd
+import pytest
+
+from life_expectancy.cleaning import Cleaner, main
+from . import DATA_DIR
+
+
+def test_run(input_df, pt_life_expectancy_expected):
+    """Test run"""
+    cleaner = Cleaner(
+        "not important",
+        "not important",
+        "PT",
+    )
+    cleaner.df = input_df
+    cleaner.run()
+    pd.testing.assert_frame_equal(
+        cleaner.df.reset_index(drop=True),
+        pt_life_expectancy_expected.reset_index(drop=True),
+        check_dtype=False,
+    )
+
+
+def test_load_data(data_file):
+    """Test load"""
+    cleaner = Cleaner(data_file, "not important", "not important")
+    cleaner.load_and_set_df()
+    pd.testing.assert_frame_equal(
+        cleaner.df,
+        pd.read_csv(data_file, sep=r"[\t,]", header=0, engine="python").reset_index(
+            drop=True
+        ),
+        check_dtype=False,
+    )
+
+
+def test_main(input_df, monkeypatch: pytest.MonkeyPatch):
+    def return_fixture(self) -> pd.DataFrame:
+        return input_df
+
+    monkeypatch.setattr(Cleaner, "_load_df", return_fixture)
+    monkeypatch.setattr(Cleaner, "save_df", lambda _: True)
+    main("PT")
```

Comment:
The patches work great, and your CleanerMock would also get the job done! 👍🏻 
But you are not testing the end result. You can have your main return either the Dataframe or the Cleaner for that.

For example, if your main returns the cleaner:

```suggestion
def test_main(input_df, pt_life_expectancy_expected, monkeypatch: pytest.MonkeyPatch):
    def return_fixture(self) -> pd.DataFrame:
        return input_df

    monkeypatch.setattr(Cleaner, "_load_df", return_fixture)
    monkeypatch.setattr(Cleaner, "save_df", lambda _: True)
    cleaner = main("PT")
    actual = cleaner.df.reset_index(drop=True)
    pd.testing.assert_frame_equal(actual, pt_life_expectancy_expected)
```